examples/stm32/nucleo-f429zi-freertos-mip/mcu.h

```diff
@@ -0,0 +1,161 @@
+// Copyright (c) 2022 Cesanta Software Limited
+// All rights reserved
+// https://www.st.com/resource/en/reference_manual/dm00031020-stm32f405-415-stm32f407-417-stm32f427-437-and-stm32f429-439-advanced-arm-based-32-bit-mcus-stmicroelectronics.pdf
+
+#pragma once
+
+#include <inttypes.h>
+#include <stdbool.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <sys/stat.h>
+
+#include "stm32f429xx.h"
+
+#define BIT(x) (1UL << (x))
+#define SETBITS(R, CLEARMASK, SETMASK) (R) = ((R) & ~(CLEARMASK)) | (SETMASK)
+#define PIN(bank, num) ((((bank) - 'A') << 8) | (num))
+#define PINNO(pin) (pin & 255)
+#define PINBANK(pin) (pin >> 8)
+
+// 6.3.3: APB1 clock <= 45MHz; APB2 clock <= 90MHz
+// 3.5.1, Table 11: configure flash latency (WS) in accordance to clock freq
+// 33.4: The AHB clock must be at least 25 MHz when Ethernet is used
+enum { APB1_PRE = 5 /* AHB clock / 4 */, APB2_PRE = 4 /* AHB clock / 2 */ };
+enum { PLL_HSI = 16, PLL_M = 8, PLL_N = 180, PLL_P = 2 };  // Run at 180 Mhz
+#define PLL_FREQ (PLL_HSI * PLL_N / PLL_M / PLL_P)
+#define FLASH_LATENCY 5
+#define FREQ (PLL_FREQ * 1000000)  // CPU frequency
+
+static inline void spin(volatile uint32_t count) {
+  while (count--) asm("nop");
+}
+
+static inline void systick_init(uint32_t ticks) {
+  if ((ticks - 1) > 0xffffff) return;  // Systick timer is 24 bit
+  SysTick->LOAD = ticks - 1;
+  SysTick->VAL = 0;
+  SysTick->CTRL = BIT(0) | BIT(1) | BIT(2);  // Enable systick
+  RCC->APB2ENR |= BIT(14);                   // Enable SYSCFG
```

Comment:
` RCC->APB2ENR |= BIT(14);`
This doesn't really belong here, and is duplicated in _main.c_

Reply:
Well.. Yes, it is duplicated in main.c but for a different reason - to enable ethernet.
Here, we enable systick. 
I'll remove the main.c line

Reply:
I beg to differ. I would remove it from here and set it at main.c

Rationale:
In main.c you do act on the SYSCFG to enable RMII, so it makes sense to enable the SYSCFG clock there.
SysTick is ARM, it does not have a clock gating and (usually) does not depend on vendor peripherals. Plus, you are acting on this _after_ setting the SysTick registers, if it were actually needed you'd have met Mr. HardFault.
From 5.2 at the datasheet:
> The RCC feeds the external clock of the Cortex System Timer (SysTick) with the AHB clock
(HCLK) divided by 8. The SysTick can work either with this clock or with the Cortex clock
(HCLK), configurable in the SysTick control and status register.

SysTick control and status register is also ARM's (SYST_CSR in Ozone), we don't act upon it and defaults to CLJSOURCE=1, receiving the Cortex clock. https://developer.arm.com/documentation/ddi0413/d/system-control/system-control-register-descriptions/systick-control-and-status-register
 
Since we calculate ticks from FREQ and they seem correct, it is receiving the Cortex clock, as usual.
Verified everything works if removing this line here and leaving it on main.c

Reply:
Done!